### PR TITLE
YLP-2697 App routing should reveal data dependencies (v1)

### PR DESCRIPTION
**Pull request by TimCanu**
## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [x] End-to-end tests successful on CI

### CODE REVIEW

**Review by celia-rouis — COMMENTED** [review]
Looks like it works 👀 🥳

File: locales/en/yourloops.json [hunk 1]
```diff
@@ -441,6 +441,7 @@
   "refresh": "Refresh",
   "remaining-time": "Remaining time",
   "remove-caregiver": "Remove caregiver",
+  "remove-member": "Remove the member {{fullName}}",
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply locales/en/yourloops.json:444]: I spotted an aria label that made no sense next to a line I modified, I figured that it was an ok opportunity to fix it

File: packages/yourloops/components/header-bars/main-header.tsx [hunk 2]
```diff
@@ -173,17 +163,17 @@ const MainHeader: FC<MainHeaderProps> = (props) => {
                   label={t('header-tab-patients')}
                   value={HcpNavigationTab.Patients}
                   onClick={() => {
-                    handleTabClick(HcpNavigationTab.Patients)
+                    navigate(`${AppUserRoute.Teams}/${teamId}/patients`)
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/components/header-bars/main-header.tsx:166]: I don't really understand this comment I think. If I understand it properly then I don't see how I can make an enum out of this as there is a dynamic value :thinking:

File: packages/yourloops/components/header-bars/main-header.tsx [hunk 3]
```diff
@@ -173,17 +163,17 @@ const MainHeader: FC<MainHeaderProps> = (props) => {
...
                   }}
                 />
-                {!TeamUtils.isPrivate(selectedTeam) &&
+                {teamId !== PRIVATE_TEAM_ID &&
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/components/header-bars/main-header.tsx:169]: I thought of it but I don't think that it is so useful (as it is a basic comparison and you'll still endup importing `TeamUtils` instead)

**Review by celia-rouis — APPROVED** [review]

**Pull request merged**